test/integ/main_test.go

```diff
@@ -736,12 +742,14 @@ func testIsIntegration(tb testing.TB) bool {
 }
 
 // Create an asymmetric signing key for use with integration tests.
-func testCreateKey(ctx context.Context, tb testing.TB, kmsClient *kms.KeyManagementClient, keyRing string) string {
+func testCreateKey(ctx context.Context, tb testing.TB, kmsClient *kms.KeyManagementClient, keyRing, primaryKeyVersion string) string {
 	tb.Helper()
 	u, err := uuid.NewUUID()
 	if err != nil {
 		tb.Fatalf("failed to create uuid : %s", err)
 	}
+	labels := make(map[string]string)
+	labels["primary"] = "ver_" + primaryKeyVersion
```

Comment:
1. It would be better to just make the map: `labels := map[string]string{"primary": "ver_"+primaryKeyVersion}`

2. Why do we need to store the primary version on a label? There's already a `primary` field on the proto which links to the specific version: https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys#CryptoKey

Reply:
Why do we need to store the primary version on a label? 
JVS use labels to store the primary version https://github.com/abcxyz/jvs/blob/23cdda185a8fc277ba798bfaf3e34543458f1934/pkg/jvscrypto/kmsutil.go#L198 instead of the `primary` field. Do we want to change it to the `primary` field now?

Reply:
We should definitely use the `primary` field in the KMS API. I'm surprised we didn't discover that in the first place.

Reply:
@raserva do you remember why we used labels to store the primary version instead of the `primary` proto field?

It would be better to use the `primary` field directly, since fetching the `cryptoKey` will then return the embedded `cyrptoKeyVersion` directly in the same API call. There's a separate API method for setting the primary version: https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys/updatePrimaryVersion

Reply:
primary isn't supported for async keys.

Reply:
https://cloud.google.com/kms/docs/key-states#symmetric_encryption

Reply:
Trying to find where that is explicitly stated, but i did try to use that field.

Reply:
This is called out in [the original PR](https://github.com/abcxyz/jvs/pull/25). 

> Unfortunately, there is no standard way in KMS to mark an asymmetric key as primary (there is a primary field, but it is unsupported for asymettric keys). Therefore, we need to store this information somewhere else. This solves this issue by utilizing [KMS key labels](https://cloud.google.com/kms/docs/creating-managing-labels) primary, new, old. This allows us to distinguish between new key versions and handle odd cases gracefully, such as rollbacks.

Reply:
Here you go. 
```
rsrv@rsrv-desktop:~$ gcloud kms keys set-primary-version <mykey> --version=1 --location=global --keyring=ci-keyring
ERROR: (gcloud.kms.keys.set-primary-version) FAILED_PRECONDITION: Keys with purpose ASYMMETRIC_SIGN do not have a primary version.
- '@type': type.googleapis.com/google.rpc.PreconditionFailure
  violations:
  - subject: <mykey>
    type: CRYPTO_SCHEME_MISMATCH
- '@type': type.googleapis.com/google.rpc.DebugInfo
  detail: '[ORIGINAL ERROR] generic::failed_precondition: Keys with purpose ASYMMETRIC_SIGN
    do not have a primary version. [google.rpc.error_details_ext] { message: "Keys
    with purpose ASYMMETRIC_SIGN do not have a primary version." details { [type.googleapis.com/google.rpc.PreconditionFailure]
    { violations { type: "CRYPTO_SCHEME_MISMATCH" subject: "<mykey>"
    } } } }'
```

Reply:
I found docs here: https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys
`Keys with [purpose](https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys#CryptoKey.FIELDS.purpose) [ENCRYPT_DECRYPT](https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys#CryptoKeyPurpose.ENUM_VALUES.ENCRYPT_DECRYPT) may have a primary. For other keys, this field will be omitted.`

Reply:
I will add a comment in setting primary method to explicitly list the reason why we don't use the `primary` field.

Reply:
Let's document that rationale inline 😄